### DISCUSSION TITLE
Lora callback reset

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#13138] [Hardware & Peripherals]
When I started with lora I used the 
```
if lora.received_packet()
```
option in a loop. This works OK but uses about 60mA because the esp32 is hammering the lora modem to see if any packets have come in. It also gives slightly less receive sensitivity because of RF noise from the esp32.

I'd prefer to use 
```
lora.receive()
```
with a callback. With this option I can put the esp32 into lightsleep & use 
```
esp32.wake_on_ext0(pin=dio_0, level=esp32.WAKEUP_ANY_HIGH)
``` 
to wake it up when the modem dio_0 pin goes high indicating a payload is available for lora.read_payload(). This draws only 22mA but the problem is the modems dio_0 pin is latched.  This isn't an issue with deepsleep after a received packet processing because the program resumes at the beginning when it wakes up & so the modem gets a fresh lora setup each time which unlatches the dio_0 pin.

My problem is I now have an application where I need to process several receives before deepsleep but I can't because the dio_0 pin is latched hi after the first receive. At the moment I'm doing a lora setup after each receive to force dio_0 low but it feels cumbersome.

I'm thinking there should be a way to unlatch the dio_0 pin after I process a receive by writing to whatever modem register is holding it high but, after a lot of googling, I can't figure out how.

Anyone else come across this problem? If so did you find a better way to reset the modems dio_0 pin without doing a full modem setup?

**Comment by davefes** [comment]
Which LoRa distro?

**Reply by kjm1102** [reply]
I'm still using the Lemariva lib on a TTGO board

**Comment by davefes** [comment]
Are you doing a lora.receive() after the first and subsequent callbacks?

**Reply by kjm1102** [reply]
yes sir
